plutus-contract/src/Plutus/Contract/Test.hs

```diff
@@ -419,22 +418,12 @@ valueAtAddress address check = TracePredicate $
 plutusValueAtAddress :: CardanoAddress -> (Plutus.Value -> Bool) -> TracePredicate
 plutusValueAtAddress addr p = valueAtAddress addr (p . fromCardanoValue)
 
--- | Get a datum of a given type 'd' out of a Transaction Output.
-getTxOutDatum ::
-  forall d.
-  (FromData d) =>
-  Ledger.CardanoTx ->
-  Ledger.TxOut ->
-  Maybe d
-getTxOutDatum tx' txOut = Ledger.txOutDatumHash txOut >>= go
-    where
-        go datumHash = Map.lookup datumHash (Ledger.getCardanoTxData tx') >>= (Ledger.getDatum >>> fromBuiltinData @d)
-
-dataAtAddress :: forall d . FromData d => CardanoAddress -> ([d] -> Bool) -> TracePredicate
+
+dataAtAddress :: CardanoAddress -> ([PV1.Datum] -> Bool) -> TracePredicate
 dataAtAddress address check = TracePredicate $
     flip postMapM (L.generalize $ Folds.utxoAtAddress address) $ \utxo -> do
       let
-        datums = mapMaybe (uncurry $ getTxOutDatum @d) $ toList utxo
+        datums = mapMaybe (\(_txId, txOut) -> Ledger.Tx.Internal.cardanoTxOutDatum txOut) $ toList utxo
```

Comment:
Please explicitly import `cardanoTxOutDatum` (or `txOutDatum` after you've renamed it) and don't use the module prefix here.

I'd also prefer to keep the original type signature of `dataAtAddress` so that the Governance example doesn't need to be changed. Then this line would become:

```haskell
datums = mapMaybe (\(_txId, txOut) -> fromBuiltinData @d . Ledger.getDatum <$> txOutDatum txOut) $ toList utxo
```